daft/analytics.py

```diff
@@ -62,21 +64,38 @@ def _build_segment_batch_payload(
     }
 
 
+def _enable_analytics() -> None:
+    global _ANALYTICS_ACTIVE
+    _ANALYTICS_ACTIVE = True
+
+
 def _post_segment_track_endpoint(payload: dict[str, Any]) -> None:
     """Posts a batch of JSON data to Segment"""
-    req = urllib.request.Request(
-        _SEGMENT_BATCH_ENDPOINT,
-        method="POST",
-        headers={
-            "Content-Type": "application/json",
-            "User-Agent": "daft-analytics",
-            "Authorization": f"Basic {base64.b64encode(f'{_WRITE_KEY}:'.encode()).decode('utf-8')}",
-        },
-        data=json.dumps(payload).encode("utf-8"),
-    )
-    resp = urllib.request.urlopen(req)
-    if resp.status != 200:
-        raise RuntimeError(f"HTTP request to segment returned status code: {resp.status}")
+    global _ANALYTICS_ACTIVE
+    if _ANALYTICS_ACTIVE:
+        req = urllib.request.Request(
+            _SEGMENT_BATCH_ENDPOINT,
+            method="POST",
+            headers={
+                "Content-Type": "application/json",
+                "User-Agent": "daft-analytics",
+                "Authorization": f"Basic {base64.b64encode(f'{_WRITE_KEY}:'.encode()).decode('utf-8')}",
+            },
+            data=json.dumps(payload).encode("utf-8"),
+        )
+        try:
+            resp = urllib.request.urlopen(
+                req,
+                timeout=1,
+            )
+        except urllib.error.URLError as error:
+            if isinstance(error.reason, socket.timeout):
+                _ANALYTICS_ACTIVE = False
+                # pass
+        if resp.status != 200:
+            raise RuntimeError(f"HTTP request to segment returned status code: {resp.status}")
```

Comment:
Any reason why we choose to only do this in the case of timeouts? Why not apply a blanket `if resp.status != 200`

Reply:
I saw that the `_flush` method always catches exceptions, logs them, and continues (without disabling the `AnalyticsClient`). I thought that we would want the `AnalyticsClient` to continue to persist through all errors except for urlopen Timeout errors.

This can be easily changed though! Let me know what you think.

Reply:
Edited to disable on any `URLError`.